vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -216,65 +209,118 @@ private void addExperimentInformationDialogListeners(ExperimentId experimentId,
+    experimentalVariablesComponent.addEditListener(
+        editEvent -> openExperimentalVariablesEditDialog());
+  }
+
+  private void deleteExistingExperimentalVariables() {
+    ExperimentId experimentId = context.experimentId().orElseThrow();
+    var result = deletionService.deleteAllExperimentalVariables(experimentId);
+    result.onError(responseCode -> {
+      throw new ApplicationException("variable deletion failed: " + responseCode);
     });
   }
 
-  private static ConfirmDialog experimentalGroupDeletionConfirmDialog() {
-    var confirmDialog = new ConfirmDialog();
-    confirmDialog.setHeader("Your experimental groups will be deleted");
-    confirmDialog.setText(
-        "Editing experimental variables requires all experimental groups to be deleted. Are you sure you want to delete them?");
-    confirmDialog.setConfirmText("Delete experimental groups");
-    confirmDialog.setCancelable(true);
-    confirmDialog.setCancelText("Abort");
-    confirmDialog.setRejectable(false);
-    return confirmDialog;
+  private void reloadExperimentInfoOnExperimentEditEvent() {
+    addListener(ExperimentEditEvent.class, event -> reloadExperimentInfo(event.experimentId()));
   }
 
-  private void reloadExperimentalVariables() {
-    experimentInformationService.find(context.experimentId().orElseThrow())
+  private void reloadExperimentInfo(ExperimentId experimentId) {
+    experimentInformationService.find(experimentId)
         .ifPresent(this::loadExperimentInformation);
   }
 
-  private void deleteExistingExperimentalVariables(ExperimentId experimentId) {
-    var result = deletionService.deleteAllExperimentalVariables(experimentId);
-    result.onError(responseCode -> {
-      throw new ApplicationException("variable deletion failed: " + responseCode, ErrorCode.GENERAL,
-          ErrorParameters.empty());
-    });
+  private void openExperimentalVariablesAddDialog() {
+    if (editVariablesNotAllowed()) {
+      return;
+    }
+    var addDialog = new ExperimentalVariablesDialog();
+    addDialog.addCancelEventListener(cancelEvent -> cancelEvent.getSource().close());
+    addDialog.addConfirmEventListener(this::onExperimentalVariablesAddConfirmed);
+    addDialog.open();
+  }
+
+  private void onExperimentalVariablesAddConfirmed(
+      ExperimentalVariablesDialog.ConfirmEvent confirmEvent) {
+    addExperimentalVariables(confirmEvent.getSource().definedVariables());
+    confirmEvent.getSource().close();
+    setContext(this.context); //reload
+    if (hasExperimentalGroups()) {
+      showSampleRegistrationPossibleNotification();
+    }
   }
 
-  private void addListenerForNewEditEvent() {
-    this.editListeners.add(event -> experimentInformationService.find(event.experimentId())
-        .ifPresent(this::loadExperimentInformation));
+  private void openExperimentalVariablesEditDialog() {
+    if (editVariablesNotAllowed()) {
+      return;
+    }
```

Comment:
I'd suggest that the notification dialog is handled outside of the boolean method

Reply:
I do not know how you would do that 😄 I am open for suggestions here.